### PR TITLE
scan_builder function added to Snapshot 

### LINKED ISSUE #206: Add a `scan_builder()` method to `Snapshot`
See https://github.com/delta-incubator/delta-kernel-rs/pull/205#discussion_r1602632346 for context

### PR DESCRIPTION
**Pull request by sherlockbeard**
Resolves #206

### CODE REVIEW

File: kernel/src/snapshot.rs [hunk 1]
```diff
@@ -248,6 +251,10 @@ impl Snapshot {
     pub fn column_mapping_mode(&self) -> ColumnMappingMode {
         self.column_mapping_mode
     }
+    /// Create a [`ScanBuilder`] for this `Snapshot`.
+    pub fn scan_builder(&self) -> ScanBuilder {
+        ScanBuilder::new(self.clone())
+    }
```
Comment (nicklan) [inline kernel/src/snapshot.rs:257]: I actually think we shouldn't `#[derive(Clone)]` and then clone ourselves. That's somewhat surprising to a user. Instead, let's implement it for `Arc`, and add an `into_scan_builder()` that consumes the snapshot:

```suggestion
    pub fn scan_builder(self: Arc<Self>) -> ScanBuilder {
        ScanBuilder::new(self)
    }
    
    pub fn into_scan_builder(self) -> ScanBuilder {
        ScanBuilder::new(self)
    }
```

You'll need to import `Arc` for this to work. Also please remove the `derives` for `Clone`

**Review by nicklan — CHANGES_REQUESTED** [review]
Thanks! Just a small suggestion on the api.

File: kernel/src/snapshot.rs (hunk 1 above)
Comment (sherlockbeard) [reply kernel/src/snapshot.rs]: ok added scan_builder, into_scan_builder

File: kernel/src/snapshot.rs [hunk 2]
```diff
@@ -248,6 +251,15 @@ impl Snapshot {
     pub fn column_mapping_mode(&self) -> ColumnMappingMode {
         self.column_mapping_mode
     }
+
+    /// Create a [`ScanBuilder`] for this `Snapshot`.
```
Comment (nicklan) [inline kernel/src/snapshot.rs:255]:
```suggestion
    /// Create a [`ScanBuilder`] for an `Arc<Snapshot>`.
```

File: kernel/src/snapshot.rs [hunk 3]
```diff
@@ -248,6 +251,15 @@ impl Snapshot {
...
+    pub fn scan_builder(self: Arc<Self>) -> ScanBuilder {
+        ScanBuilder::new(self)
+    }
+
+    pub fn into_scan_builder(self) -> ScanBuilder {
```
Comment (nicklan) [inline kernel/src/snapshot.rs:260]:
```suggestion
    /// Consume this `Snapshot` to create a [`ScanBuilder`]
    pub fn into_scan_builder(self) -> ScanBuilder {
```

**Review by nicklan — APPROVED** [review]
lgtm, just please two small documentation fixes! Thanks!

**Review by hntd187 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**